### DISCUSSION TITLE
Requirements on a `prisma migrate reset`

### DISCUSSION THREAD

**Discussion by lostadam** [prisma/prisma#17791] [Q&A]
Hi everyone! I've a quick question. We run a docker-compose setup, with something like the following services:
- `postgres`
- `crud-api`
- `prisma-container`

The `crud-api` uses prisma with the (single instance pattern)[https://www.prisma.io/docs/guides/performance-and-optimization/connection-management#re-using-a-single-prismaclient-instance].

Our seed scripts involve two steps:
1. Directly writing to the prisma instance to table `A`
2. Checking a _different_ database to see if any instances should be added to table `B`. If so, hitting the `crud-api` on a `POST /b` route.

The issue is that on the `POST /b`, if there _was_ something to add, we were getting the following error:

```
connectorError(ConnectorError { user_facing_error: None, kind: QueryError(Error { kind: Db, cause: Some(DbError { severity: ERROR, parsed_severity: Some(Error), code: SqlState(EXX000), message: cache lookup failed for type 16820, detail: None, hint: None, position: None, where_: Some(unnamed portal parameter $1), schema: None, table: None, column: None, datatype: None, constraint: None, file: Some(lsyscache.c), line: Some(2904), routine: Some(getTypeBinaryInputInfo) }) }), transient: false })
```

Postgres would spit out a similar `cache lookup failed for type 16820` somewhere in the `docker compose logs`, and things would fail.

After a ton of hacking around, I added a `/prisma` route to `crud-api`, that looks like the following:

```
router.post("/prisma", async () => {
  await prisma.$disconnect()
  await prisma.$connect()
})
```

When I call this before `POST /b`, things work smoothly.

Just wanted to bring this to you all's attention, as I didn't know that we would have to reset connections on any `reset`. Maybe adding this to the docs could be helpful for someone that comes across something similar!

**Comment by lostadam** [comment]
Hey building on this, I think this is actually fairly undesirable behavior.

Our _actual_ setup has 4/5 services that depend on `prisma`, and not all can have a `/prisma` hacked in (e.g. a message queue). Without knowing the internals of how `npx prisma migrate reset` operates, I think it would be desirable to have it clear out any caching mechanism that the underlying db is doing in relation to any prisma relations.

**Reply by janpio** [reply]
This is obviously not intended behavior.

What is the connection between the problem you describe, which is something going wrong in Prisma Client, and `migrate reset` though?

How do you "Check[...] a different database to see if any instances should be added to table B"? Also using Prisma with another Prisma Client?

More general: Is this `cache lookup failed for type 16820` problem reproducible? We sometimes get reports of this but could never reproduce this. If so, your project might be highly valuable for us to understand what is going on here.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
